### PR TITLE
Update E2E tests to run Kubernetes 1.31

### PR DESCRIPTION
**Pull request by thunderboltsid**
* Update default e2e os images to rocky9 to be in line with NKP defaults
* Update kubevip on e2e to 0.8.2 to align with templates
* Update CAPX version for clusterctl-upgrade test to last v1.5.x to test clusterctl-upgrade for v1.6.0

### CODE REVIEW

**Review by adiantum — APPROVED** [review]
lgtm

File: test/e2e/clusterctl_upgrade_test.go [hunk 1]
```diff
@@ -26,20 +26,19 @@ import (
...
-	"github.com/blang/semver/v4"
-	"github.com/nutanix-cloud-native/cluster-api-provider-nutanix/test/e2e/log"
-
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 	bootstrapv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1beta1"
 	yaml "sigs.k8s.io/cluster-api/cmd/clusterctl/client/yamlprocessor"
-	capi_e2e "sigs.k8s.io/cluster-api/test/e2e"
+	capie2e "sigs.k8s.io/cluster-api/test/e2e"
```
Comment (deepakm-ntnx) [inline test/e2e/clusterctl_upgrade_test.go:37]: whats the context for this change in naming

Comment (deepakm-ntnx) [reply test/e2e/clusterctl_upgrade_test.go:37]: per @thunderboltsid to fix lint related issues

File: test/e2e/config/nutanix.yaml [hunk 2]
```diff
@@ -153,26 +153,24 @@ variables:
...
   NUTANIX_PRISM_ELEMENT_CLUSTER_NAME: ""
-  NUTANIX_MACHINE_TEMPLATE_IMAGE_NAME: "ubuntu-2204-kube-v1.30.2.qcow2"
-  NUTANIX_MACHINE_TEMPLATE_IMAGE_NAME_v1_27: "ubuntu-2204-kube-v1.27.13.qcow2"
-  NUTANIX_MACHINE_TEMPLATE_IMAGE_NAME_v1_28: "ubuntu-2204-kube-v1.28.9.qcow2"
-  NUTANIX_MACHINE_TEMPLATE_IMAGE_NAME_v1_29: "ubuntu-2204-kube-v1.29.6.qcow2"
-  NUTANIX_MACHINE_TEMPLATE_IMAGE_NAME_v1_30: "ubuntu-2204-kube-v1.30.2.qcow2"
+  NUTANIX_MACHINE_TEMPLATE_IMAGE_NAME: "rockylinux-9-kube-v1.31.0.qcow2"
+  NUTANIX_MACHINE_TEMPLATE_IMAGE_NAME_v1_28: "rockylinux-9-kube-v1.28.13.qcow2"
+  NUTANIX_MACHINE_TEMPLATE_IMAGE_NAME_v1_29: "rockylinux-9-kube-v1.29.8.qcow2"
+  NUTANIX_MACHINE_TEMPLATE_IMAGE_NAME_v1_30: "rockylinux-9-kube-v1.30.4.qcow2"
```
Comment (deepakm-ntnx) [inline test/e2e/config/nutanix.yaml:169]: we should add NUTANIX_MACHINE_TEMPLATE_IMAGE_NAME_v1_31 as well or planning to add when we add test?

**Review by deepakm-ntnx — APPROVED** [review]

File: test/e2e/config/nutanix.yaml (hunk 2 above)
Comment (thunderboltsid) [reply test/e2e/config/nutanix.yaml:169]: Good point. It's basically same as the default i.e. `NUTANIX_MACHINE_TEMPLATE_IMAGE_NAME`. So, we can add it later.

**Pull request merged**